Docs fail to build: Semver version 2024.1 is older than 2 years

Hi,

i am currently unable to build authentik after a couple months of not upgrading (2025.10.2 -> 2025.12.1-dependency-fix). It seems the docs are failing to build.

```
goauthentik-docs> npm info using npm@11.6.2
goauthentik-docs> npm info using node@v24.12.0
goauthentik-docs> npm info config found workspace root at /build/website
goauthentik-docs> npm verbose title npm run build:docusaurus
goauthentik-docs> npm verbose argv "run" "build:docusaurus"
goauthentik-docs> npm verbose logfile logs-max:10 dir:/build/tmp.HXOMCKHlCi/.npm/_logs/2026-02-03T00_23_24_153Z-
goauthentik-docs> npm verbose logfile /build/tmp.HXOMCKHlCi/.npm/_logs/2026-02-03T00_23_24_153Z-debug-0.log
goauthentik-docs>
goauthentik-docs> > @goauthentik/docs-topics@0.0.0 build:docusaurus
goauthentik-docs> > docusaurus build
goauthentik-docs>
goauthentik-docs> [INFO] [en] Creating an optimized production build...
goauthentik-docs> 🚀 ak-releases-plugin loaded
goauthentik-docs> 🚀 ak-redirects-plugin loaded
goauthentik-docs> ✅ releases.gen.json generated
                  ● Client ██████████████████████████████████████████████████ (93%) sealing after
goauthentik-docs> ● Server ██████████████████████████████████████████████████ (93%) sealing after [ERROR] Client bundle compiled with errors therefore further build is impossible.
goauthentik-docs> × Module build failed (from ../node_modules/@docusaurus/mdx-loader/lib/index.js):
goauthentik-docs>   ╰─▶   × Error: MDX compilation failed for file "/build/website/docs/add-secure-apps/flows-stages/stages/identification/index.mdx"
goauthentik-docs>         │ Cause: Semver version 2024.1 is older than 2 years.
goauthentik-docs>         │ Details:
goauthentik-docs>         │ Error: Semver version 2024.1 is older than 2 years.
goauthentik-docs>         │     at assertVersionSupported (file:///build/website/docusaurus-theme/releases/version.mjs:28:15)
goauthentik-docs>         │     at file:///build/website/docusaurus-theme/remark/version-directive.mjs:48:13
goauthentik-docs>         │     at overload (file:///build/website/node_modules/unist-util-visit/lib/index.js:311:12)
goauthentik-docs>         │     at node (textDirective<ak-version>) (file:///build/website/node_modules/unist-util-visit-parents/lib/index.js:347:27)
goauthentik-docs>         │     at node (heading) (file:///build/website/node_modules/unist-util-visit-parents/lib/index.js:364:61)
goauthentik-docs>         │     at node (root) (file:///build/website/node_modules/unist-util-visit-parents/lib/index.js:364:61)
goauthentik-docs>         │     at visitParents (file:///build/website/node_modules/unist-util-visit-parents/lib/index.js:305:31)
goauthentik-docs>         │     at visit (file:///build/website/node_modules/unist-util-visit/lib/index.js:302:3)
goauthentik-docs>         │     at file:///build/website/docusaurus-theme/remark/version-directive.mjs:34:9
goauthentik-docs>         │     at wrapped (file:///build/website/node_modules/trough/lib/index.js:160:27)
goauthentik-docs>         │     at next (file:///build/website/node_modules/trough/lib/index.js:86:23)
goauthentik-docs>         │     at done (file:///build/website/node_modules/trough/lib/index.js:194:7)
goauthentik-docs>         │     at then (file:///build/website/node_modules/trough/lib/index.js:204:5)
goauthentik-docs>         │     at wrapped (file:///build/website/node_modules/trough/lib/index.js:181:9)
goauthentik-docs>         │     at next (file:///build/website/node_modules/trough/lib/index.js:86:23)
goauthentik-docs>         │     at Object.run (file:///build/website/node_modules/trough/lib/index.js:57:5)
goauthentik-docs>         │     at executor (file:///build/website/node_modules/unified/lib/index.js:901:20)
goauthentik-docs>         │     at apply.run (file:///build/website/node_modules/unified/lib/index.js:884:19)
goauthentik-docs>         │     at executor (file:///build/website/node_modules/unified/lib/index.js:737:12)
goauthentik-docs>         │     at new Promise (<anonymous>)
goauthentik-docs>         │     at apply.process (file:///build/website/node_modules/unified/lib/index.js:720:47)
goauthentik-docs>         │     at Object.process (/build/website/node_modules/@docusaurus/mdx-loader/lib/processor.js:150:37)
goauthentik-docs>         │     at compileToJSX (/build/website/node_modules/@docusaurus/mdx-loader/lib/utils.js:77:32)
goauthentik-docs>         │     at async loadMDX (/build/website/node_modules/@docusaurus/mdx-loader/lib/loader.js:22:20)
goauthentik-docs>         │     at compileToJSX (/build/website/node_modules/@docusaurus/mdx-loader/lib/utils.js:93:15)
goauthentik-docs>         │     at async loadMDX (/build/website/node_modules/@docusaurus/mdx-loader/lib/loader.js:22:20)
goauthentik-docs>
goauthentik-docs>
goauthentik-docs> --------------------------
goauthentik-docs>
goauthentik-docs> × Module build failed (from ../node_modules/@docusaurus/mdx-loader/lib/index.js):
goauthentik-docs>   ╰─▶   × Error: MDX compilation failed for file "/build/website/docs/install-config/configuration/configuration.mdx"
goauthentik-docs>         │ Cause: Semver version 2024.1 is older than 2 years.
goauthentik-docs>         │ Details:
goauthentik-docs>         │ Error: Semver version 2024.1 is older than 2 years.
goauthentik-docs>         │     at assertVersionSupported (file:///build/website/docusaurus-theme/releases/version.mjs:28:15)
goauthentik-docs>         │     at file:///build/website/docusaurus-theme/remark/version-directive.mjs:48:13
goauthentik-docs>         │     at overload (file:///build/website/node_modules/unist-util-visit/lib/index.js:311:12)
goauthentik-docs>         │     at node (textDirective<ak-version>) (file:///build/website/node_modules/unist-util-visit-parents/lib/index.js:347:27)
goauthentik-docs>         │     at node (heading) (file:///build/website/node_modules/unist-util-visit-parents/lib/index.js:364:61)
goauthentik-docs>         │     at node (root) (file:///build/website/node_modules/unist-util-visit-parents/lib/index.js:364:61)
goauthentik-docs>         │     at visitParents (file:///build/website/node_modules/unist-util-visit-parents/lib/index.js:305:31)
goauthentik-docs>         │     at visit (file:///build/website/node_modules/unist-util-visit/lib/index.js:302:3)
goauthentik-docs>         │     at file:///build/website/docusaurus-theme/remark/version-directive.mjs:34:9
goauthentik-docs>         │     at wrapped (file:///build/website/node_modules/trough/lib/index.js:160:27)
goauthentik-docs>         │     at next (file:///build/website/node_modules/trough/lib/index.js:86:23)
goauthentik-docs>         │     at done (file:///build/website/node_modules/trough/lib/index.js:194:7)
goauthentik-docs>         │     at then (file:///build/website/node_modules/trough/lib/index.js:204:5)
goauthentik-docs>         │     at wrapped (file:///build/website/node_modules/trough/lib/index.js:181:9)
goauthentik-docs>         │     at next (file:///build/website/node_modules/trough/lib/index.js:86:23)
goauthentik-docs>         │     at Object.run (file:///build/website/node_modules/trough/lib/index.js:57:5)
goauthentik-docs>         │     at executor (file:///build/website/node_modules/unified/lib/index.js:901:20)
goauthentik-docs>         │     at apply.run (file:///build/website/node_modules/unified/lib/index.js:884:19)
goauthentik-docs>         │     at executor (file:///build/website/node_modules/unified/lib/index.js:737:12)
goauthentik-docs>         │     at new Promise (<anonymous>)
goauthentik-docs>         │     at apply.process (file:///build/website/node_modules/unified/lib/index.js:720:47)
goauthentik-docs>         │     at Object.process (/build/website/node_modules/@docusaurus/mdx-loader/lib/processor.js:150:37)
goauthentik-docs>         │     at compileToJSX (/build/website/node_modules/@docusaurus/mdx-loader/lib/utils.js:77:32)
goauthentik-docs>         │     at async loadMDX (/build/website/node_modules/@docusaurus/mdx-loader/lib/loader.js:22:20)
goauthentik-docs>         │     at compileToJSX (/build/website/node_modules/@docusaurus/mdx-loader/lib/utils.js:93:15)
goauthentik-docs>         │     at async loadMDX (/build/website/node_modules/@docusaurus/mdx-loader/lib/loader.js:22:20)
goauthentik-docs>
npm verbose stack Error: command failed
goauthentik-docs> npm verbose stack     at promiseSpawn (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:22:22)
goauthentik-docs> npm verbose stack     at spawnWithShell (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:124:10)
goauthentik-docs> npm verbose stack     at promiseSpawn (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:12:12)
goauthentik-docs> npm verbose stack     at runScriptPkg (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/run-script-pkg.js:79:13)
goauthentik-docs> npm verbose stack     at runScript (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/run-script.js:9:12)
goauthentik-docs> npm verbose stack     at #run (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/lib/commands/run.js:135:13)
goauthentik-docs> npm verbose stack     at RunScript.execWorkspaces (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/lib/commands/run.js:67:24)
goauthentik-docs> npm verbose stack     at async Npm.exec (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/lib/npm.js:208:9)
goauthentik-docs> npm verbose stack     at async module.exports (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/lib/cli/entry.js:67:5)
npm verbose pkgid @goauthentik/docs-topics@0.0.0
npm error Lifecycle script `build:docusaurus` failed with error:
npm error code 1
npm error path /build/website/docs
npm error workspace @goauthentik/docs-topics@0.0.0
npm error location /build/website/docs
npm error command failed
npm error command sh -c docusaurus build
npm verbose cwd /build/website/docs
npm verbose os Linux 6.12.68
npm verbose node v24.12.0
npm verbose npm  v11.6.2
npm verbose exit 1
npm verbose code 1
ERROR: "build:docusaurus" exited with 1.
npm verbose stack Error: command failed
goauthentik-docs> npm verbose stack     at promiseSpawn (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:22:22)
goauthentik-docs> npm verbose stack     at spawnWithShell (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:124:10)
goauthentik-docs> npm verbose stack     at promiseSpawn (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:12:12)
goauthentik-docs> npm verbose stack     at runScriptPkg (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/run-script-pkg.js:79:13)
goauthentik-docs> npm verbose stack     at runScript (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/run-script.js:9:12)
goauthentik-docs> npm verbose stack     at #run (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/lib/commands/run.js:135:13)
goauthentik-docs> npm verbose stack     at RunScript.execWorkspaces (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/lib/commands/run.js:67:24)
goauthentik-docs> npm verbose stack     at async Npm.exec (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/lib/npm.js:208:9)
goauthentik-docs> npm verbose stack     at async module.exports (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/lib/cli/entry.js:67:5)
npm verbose pkgid @goauthentik/docs-topics@0.0.0
npm error Lifecycle script `build` failed with error:
npm error code 1
npm error path /build/website/docs
npm error workspace @goauthentik/docs-topics@0.0.0
npm error location /build/website/docs
npm error command failed
npm error command sh -c run-s build:docusaurus build:copy
npm verbose cwd /build/website
npm verbose os Linux 6.12.68
npm verbose node v24.12.0
npm verbose npm  v11.6.2
npm verbose exit 1
npm verbose code 1
npm verbose cwd /build/website
npm verbose os Linux 6.12.68
npm verbose node v24.12.0
npm verbose npm  v11.6.2
npm verbose exit 1
npm verbose code 1

error: builder for '/nix/store/ryhglr8lla90269fx6xq53fqffm0cya7-goauthentik-docs-2025.12.1.drv' failed with exit code 1;
       last 25 log lines:
       > npm verbose stack     at #run (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/lib/commands/run.js:135:13)
       > npm verbose stack     at RunScript.execWorkspaces (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/lib/commands/run.js:67:24)
       > npm verbose stack     at async Npm.exec (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/lib/npm.js:208:9)
       > npm verbose stack     at async module.exports (/nix/store/cikdc61gfwvdma6y0p9b5d5d448aqcv6-nodejs-24.12.0/lib/node_modules/npm/lib/cli/entry.js:67:5)
npm verbose pkgid @goauthentik/docs-topics@0.0.0
npm error Lifecycle script `build` failed with error:
npm error code 1
npm error path /build/website/docs
npm error workspace @goauthentik/docs-topics@0.0.0
npm error location /build/website/docs
npm error command failed
npm error command sh -c run-s build:docusaurus build:copy
npm verbose cwd /build/website
npm verbose os Linux 6.12.68
npm verbose node v24.12.0
npm verbose npm  v11.6.2
npm verbose exit 1
npm verbose code 1
npm verbose cwd /build/website
npm verbose os Linux 6.12.68
npm verbose node v24.12.0
npm verbose npm  v11.6.2
npm verbose exit 1
npm verbose code 1
```

--- a/components/0002-admin-file-dir-doesn-t-have-to-be-a-mountpoint.patch
+++ b/components/0002-admin-file-dir-doesn-t-have-to-be-a-mountpoint.patch
@@ -1,20 +1,20 @@
-From 2f51711b64204d090ad8cd6b2ef19fd11a1a6469 Mon Sep 17 00:00:00 2001
+From 18b41fcb663a11fcd8ccc27bd951cc150456291b Mon Sep 17 00:00:00 2001
 From: Maximilian Bosch <maximilian@mbosch.me>
 Date: Fri, 16 Jan 2026 21:50:11 +0100
-Subject: [PATCH 2/2] admin: file dir doesn't have to be a mountpoint
+Subject: [PATCH] admin: file dir doesn't have to be a mountpoint
 
 ---
  authentik/admin/files/backends/file.py | 1 -
  1 file changed, 1 deletion(-)
 
 diff --git a/authentik/admin/files/backends/file.py b/authentik/admin/files/backends/file.py
-index 7858ed5e9b..8a6d55ce64 100644
+index 6d60d5f236..720223231f 100644
 --- a/authentik/admin/files/backends/file.py
 +++ b/authentik/admin/files/backends/file.py
-@@ -47,7 +47,6 @@ class FileBackend(ManageableBackend):
-     def manageable(self) -> bool:
+@@ -52,7 +52,6 @@ class FileBackend(ManageableBackend):
+         # file creation on fresh installs.
          return (
-             self.base_path.exists()
+             self._base_dir.exists()
 -            and (self._base_dir.is_mount() or (self._base_dir / self.usage.value).is_mount())
              or (settings.DEBUG or settings.TEST)
          )

--- a/components/client-go.nix
+++ b/components/client-go.nix
@@ -1,0 +1,20 @@
+{
+  authentik-src,
+  authentik-go,
+  openapi-generator-cli,
+  runCommand,
+}:
+
+runCommand "go-client-code" {
+  nativeBuildInputs = [
+    openapi-generator-cli
+  ];
+} ''
+  cp --no-preserve=mode -vr ${authentik-go}/ $out/
+  cp -vr ${authentik-src}/schema.yml $out/
+  pushd $out &>/dev/null
+    substituteInPlace config.yaml \
+      --replace-fail "templateDir: /local/templates/" "templateDir: ./templates/"
+    openapi-generator-cli generate -i schema.yml -g go -o . -c config.yaml
+  popd &>/dev/null
+''

--- a/components/gopkgs.nix
+++ b/components/gopkgs.nix
@@ -8,10 +8,12 @@
   guacamole-server,
   stdenv,
   patches,
+  generatedGoClient,
 }:
 
 let
   guacamoleAvailable = lib.meta.availableOn stdenv.hostPlatform guacamole-server;
+
 in
 buildGo125Module {
   pname = "authentik-gopkgs";
@@ -21,6 +23,9 @@ buildGo125Module {
     sed -i"" -e 's,./web/dist/,${authentikComponents.frontend}/dist/,' web/static.go
     sed -i"" -e 's,./web/dist/,${authentikComponents.frontend}/dist/,' internal/web/static.go
     sed -i"" -e 's,./lifecycle/gunicorn.conf.py,${authentikComponents.staticWorkdirDeps}/lifecycle/gunicorn.conf.py,' internal/gounicorn/gounicorn.go
+    cp --no-preserve=mode -vr ${generatedGoClient} gen-go-api
+    echo "replace goauthentik.io/api/v3 => ./gen-go-api" >>go.mod
+    go mod edit -require=goauthentik.io/api/v3@v3.0.0
   '' + lib.optionalString guacamoleAvailable ''
     substituteInPlace internal/outpost/rac/guacd.go \
       --replace-fail '/opt/guacamole/sbin/guacd' \
@@ -43,6 +48,7 @@ buildGo125Module {
           "/web/security.txt"
           "go.mod"
           "go.sum"
+          "schema.yml"
         ])
       ))
     );
@@ -63,7 +69,7 @@ buildGo125Module {
   ] ++ lib.optionals guacamoleAvailable [
     "cmd/rac"
   ];
-  vendorHash = "sha256-u/kAqDCeWHPaw/0+lQ9U6/pHSgdANOeflQLVgUV64Vs=";
+  vendorHash = "sha256-meqQHr5U0/6nc8fv1b4SFo2ohFcwSeAmSB1eLKsO9nc=";
   nativeBuildInputs = [ makeWrapper ];
   doCheck = false;
   postInstall = ''

--- a/flake.nix
+++ b/flake.nix
@@ -42,8 +42,11 @@
     };
     authentik-src = {
       # change version string in outputs as well when updating
-      #url = "github:goauthentik/authentik/version/2025.12.1";
-      url = "github:ma27/authentik/2025.12.1-dependency-fix";
+      url = "github:goauthentik/authentik/version/2025.12.3";
+      flake = false;
+    };
+    authentik-go = {
+      url = "github:goauthentik/client-go";
       flake = false;
     };
   };
@@ -54,6 +57,7 @@
       flake-parts,
       napalm,
       authentik-src,
+      authentik-go,
       uv2nix,
       pyproject-build-systems,
       pyproject-nix,
@@ -68,7 +72,7 @@
         ...
       }:
       let
-        authentik-version = "2025.12.1"; # to pass to the drvs of some components
+        authentik-version = "2025.12.3"; # to pass to the drvs of some components
       in
       {
         systems = import inputs.systems;
@@ -127,6 +131,8 @@
                   manage = final.callPackage ./components/manage.nix { };
                 };
 
+                generatedGoClient = final.callPackage ./components/client-go.nix { };
+
                 # for uv2nix
                 pythonOverlay = final.callPackage ./components/python-overrides.nix { };
 
@@ -136,6 +142,7 @@
 
                 inherit
                   authentik-src
+                  authentik-go
                   authentik-version
                   buildNapalmPackage
                   uv2nix

